src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/edit_index_pattern.js

```diff
@@ -39,132 +44,196 @@ import { I18nContext } from 'ui/i18n';
-              );
-              $scope.refreshFilters();
-              $scope.$apply();
-            }}
-          />
-        </I18nContext>,
-        node
-      );
-    });
-  } else {
-    destroySourceFiltersTable();
-  }
+const TAB_INDEXED_FIELDS = 'indexedFields';
+const TAB_SCRIPTED_FIELDS = 'scriptedFields';
+const TAB_SOURCE_FILTERS = 'sourceFilters';
+
+function updateSourceFiltersTable($scope) {
+  $scope.$$postDigest(() => {
+    const node = document.getElementById(REACT_SOURCE_FILTERS_DOM_ELEMENT_ID);
+    if (!node) {
+      return;
+    }
+
+    render(
+      <I18nContext>
+        <SourceFiltersTable
+          indexPattern={$scope.indexPattern}
+          filterFilter={$scope.fieldFilter}
+          fieldWildcardMatcher={$scope.fieldWildcardMatcher}
+          onAddOrRemoveFilter={() => {
+            $scope.editSections = $scope.editSectionsProvider(
+              $scope.indexPattern,
+              $scope.fieldFilter,
+              $scope.indexPatternListProvider
+            );
+            $scope.refreshFilters();
+            $scope.$apply();
+          }}
+        />
+      </I18nContext>,
+      node
+    );
+  });
 }
 
 function destroySourceFiltersTable() {
   const node = document.getElementById(REACT_SOURCE_FILTERS_DOM_ELEMENT_ID);
   node && unmountComponentAtNode(node);
 }
 
-function updateScriptedFieldsTable($scope, $state) {
-  if ($state.tab === 'scriptedFields') {
-    $scope.$$postDigest(() => {
-      const node = document.getElementById(REACT_SCRIPTED_FIELDS_DOM_ELEMENT_ID);
-      if (!node) {
-        return;
-      }
-
-      render(
-        <I18nContext>
-          <ScriptedFieldsTable
-            indexPattern={$scope.indexPattern}
-            fieldFilter={$scope.fieldFilter}
-            scriptedFieldLanguageFilter={$scope.scriptedFieldLanguageFilter}
-            helpers={{
-              redirectToRoute: (obj, route) => {
-                $scope.kbnUrl.redirectToRoute(obj, route);
-                $scope.$apply();
-              },
-              getRouteHref: (obj, route) => $scope.kbnUrl.getRouteHref(obj, route),
-            }}
-            onRemoveField={() => {
-              $scope.editSections = $scope.editSectionsProvider(
-                $scope.indexPattern,
-                $scope.fieldFilter,
-                $scope.indexPatternListProvider
-              );
-              $scope.refreshFilters();
+function updateScriptedFieldsTable($scope) {
+  $scope.$$postDigest(() => {
+    const node = document.getElementById(REACT_SCRIPTED_FIELDS_DOM_ELEMENT_ID);
+    if (!node) {
+      return;
+    }
+
+    render(
+      <I18nContext>
+        <ScriptedFieldsTable
+          indexPattern={$scope.indexPattern}
+          fieldFilter={$scope.fieldFilter}
+          scriptedFieldLanguageFilter={$scope.scriptedFieldLanguageFilter}
+          helpers={{
+            redirectToRoute: (obj, route) => {
+              $scope.kbnUrl.changeToRoute(obj, route);
```

Comment:
What's the reason for using change instead of redirect?

Reply:
By adding `state_sync` utility we also fixed browser back / forward navigation when navigating around tabs. 
And changing here from redirect to change also improves back navigation.